pyi.py

```diff
@@ -205,18 +206,19 @@ def visit_BinOp(self, node: ast.BinOp) -> None:
     def visit_Subscript(self, node: ast.Subscript) -> None:
         self.generic_visit(node)
 
+        if not (isinstance(node.value, ast.Name) and node.value.id == "Union"):
+            return
+
         # Union[str, int] parses differently depending on python versions:
         # Before 3.9:     Subscript(value=Name(id='Union'), slice=Index(value=Tuple(...)))
         # 3.9 and newer:  Subscript(value=Name(id='Union'), slice=Tuple(...))
-        if isinstance(node.value, ast.Name) and node.value.id == "Union":
-            if sys.version_info >= (3, 9):
-                if isinstance(node.slice, ast.Tuple):
-                    self._check_union_members(node.slice.elts)
-            else:
-                if isinstance(node.slice, ast.Index) and isinstance(
-                    node.slice.value, ast.Tuple
-                ):
-                    self._check_union_members(node.slice.value.elts)
+        if sys.version_info >= (3, 9):
+            if isinstance(node.slice, ast.Tuple):
+                self._check_union_members(node.slice.elts)
+        elif isinstance(node.slice, ast.Index) and isinstance(
```

Comment:
I actually like the old version better here because it's more symmetrical: we have one top-level block for 3.9+ and one for <3.9, and within each block we have an AST type check. Now, instead we have an `if` and an `elif` at the same level that perform very different checks.

Reply:
Good point; I'll revise